Data/Functor/Foldable.hs

```diff
@@ -693,10 +710,10 @@ zygoHistoPrepro
   :: (Corecursive t, Recursive t)
   => (Base t b -> b)
   -> (forall c. Base t c -> Base t c)
-  -> (Base t (EnvT b (Cofree (Base t)) a) -> a)
+  -> (Base t (b, Cofree (Base t) a) -> a)
   -> t
   -> a
-zygoHistoPrepro f g t = gprepro (distZygoT f distHisto) g t
+zygoHistoPrepro f g t = gprepro (gatherZygoT f gatherHisto) g t
```

Comment:
The behaviour of `zygoHistoPrepro` has changed; it was using the problematic `distZygoT f distHisto` I argued against in #49, and now it's using the fixed version which passes the sanity check I gave there.